ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/vxlan-interface.j2

```diff
@@ -4,19 +4,27 @@
 ## VXLAN Interface
 
 ### VXLAN Interface Summary
-{%     if vxlan_interface.Vxlan1.vxlan.source_interface is arista.avd.defined %}
 
-#### Source Interface: {{ vxlan_interface.Vxlan1.vxlan.source_interface }}
+| Settings | Values |
+| -------- | ------ |
+{%     if vxlan_interface.Vxlan1.vxlan.source_interface is arista.avd.defined %}
+| Source Interface | {{ vxlan_interface.Vxlan1.vxlan.source_interface }} |
 {%     endif %}
 {%     if vxlan_interface.Vxlan1.vxlan.mlag_source_interface is arista.avd.defined %}
-
-#### MLAG Source Interface: {{ vxlan_interface.Vxlan1.vxlan.mlag_source_interface }}
+| MLAG Source Interface | {{ vxlan_interface.Vxlan1.vxlan.mlag_source_interface }} |
 {%     endif %}
-
-#### UDP port: {{ vxlan_interface.Vxlan1.vxlan.udp_port | arista.avd.default('4789') }}
+| UDP port | {{ vxlan_interface.Vxlan1.vxlan.udp_port | arista.avd.default('4789') }} |
 {%     if vxlan_interface.Vxlan1.vxlan.virtual_router_encapsulation_mac_address is arista.avd.defined %}
-
-#### EVPN MLAG Shared Router MAC : {{ vxlan_interface.Vxlan1.vxlan.virtual_router_encapsulation_mac_address }}
+| EVPN MLAG Shared Router MAC | {{ vxlan_interface.Vxlan1.vxlan.virtual_router_encapsulation_mac_address }} |
+{%     endif %}
+{%     if vxlan_interface.Vxlan1.vxlan.flood_vtep_learned_data_plane is arista.avd.defined(true) %}
+| VXLAN flood-lists learning from data-plane | Enabled |
+{%     endif %}
+{%     if vxlan_interface.Vxlan1.vxlan.qos.dscp_propagation_encapsulation is arista.avd.defined(true) %}
+| Qos dscp propagation encapsulation | Enabled |
+{%     endif %}
+{%     if vxlan_interface.Vxlan1.vxlan.qos.map_dscp_to_traffic_class_decapsulation is arista.avd.defined(true) %}
```

Comment:
Eos template covers both true and false for these options.
Maybe we should start using the `ternary` [filter](https://ansible-docs.readthedocs.io/zh/stable-2.0/rst/playbooks_filters.html#other-useful-filters).

Reply:
It is still confusing to me sometimes. I covered the false option in eos template just as a precautionary method. But that **no** is the default option. So I added only if they enable it.

We can keep the default value as False here but not all platforms support this knob. So I did this amid lot of confusion on what to do.

Reply:
If people add the key - even setting what we believe is the default - we should render config and documentation. Defaults can be affected by other settings, platforms, versions, so we should avoid judging that here.

Reply:
Ternary is useful for comparing strings. But there are some complications when it comes to true or false input.

Like->
```
a: true

{{ a | ternary('Enabled', 'Disabled') }} --> Enabled
```

```
a: false

{{ a | ternary('Enabled', 'Disabled') }} --> Disabled
```

```
a:

{{ a | ternary('Enabled', 'Disabled') }} --> Disabled
```

Reply:
Use this link to [test](https://ansible.sivel.net/test/) ansible code

Reply:
Ternary will be useful for writing default values. But in cli_config_gen we want to check that user sends only true or only false. In these cases I am not sure how to work with ternary.

Reply:
not a problem here, since we still have the `if` above with `is arista.avd.defined`

Reply:
a: not

{{ a | ternary('Enabled', 'Disabled') }} --> Enabled

Reply:
```
a: no

{{ a | ternary('Enabled', 'Disabled') }} --> Disabled
```

Reply:
Done, implemented ternary.